semconvgen: unable to access specifcation data due to incorrect selinux file labels

Enforced selinux policies can cause issues when files get shared with a containerized process. Since file permissions are reflected by this labels[0]. Docker/Podman have the ability to relabel[1] shared files.

0: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/security-enhanced_linux/sect-security-enhanced_linux-working_with_selinux-selinux_contexts_labeling_files
1: https://docs.docker.com/storage/bind-mounts/#configure-the-selinux-label

---

Generating semconv 1.16.0 on fedora 37.

```bash
$ make gensemconv SPECPATH=../opentelemetry-specification SPECTAG=v1.16.0
Generating semantic convention constants from specification version v1.16.0 at ../opentelemetry-specification
semconvgen -o semconv/v1.16.0 -t semconv/template.j2 -s v1.16.0 -i ../opentelemetry-specification/semantic_conventions/. --only=resource -p conventionType=resource -f generated_resource.go
panic: unable to render template: exit status 2
usage: gen-semconv [-h] [--debug]
                   [--only {span,resource,event,metric,units,scope}]
                   [--yaml-root folder] [--exclude EXCLUDE]
                   [YAML_FILE [YAML_FILE ...]] {code,markdown} ...
gen-semconv: error: No semantic convention model found!


goroutine 1 [running]:
main.main()
	/home/frzifus/git/go/observability/opentelemetry-go-build-tools/semconvgen/generator.go:62 +0x48e
make: *** [Makefile:290: gensemconv] Fehler 2

```

---

Details: https://github.com/open-telemetry/opentelemetry-go-build-tools/pull/217#discussion_r1063398151

--- a/.chloggen/fix_semconvgen.yaml
+++ b/.chloggen/fix_semconvgen.yaml
@@ -1,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. crosslink)
+component: semconvgen
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add docker/podman mount option to modify selinux labels of shared host files.
+
+# One or more tracking issues related to the change
+issues: [222]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

--- a/semconvgen/generator.go
+++ b/semconvgen/generator.go
@@ -164,7 +164,7 @@ func render(cfg config) error {
 
 	args := []string{
 		"run", "--rm",
-		"-v", fmt.Sprintf("%s:/data", tmpDir),
+		"-v", fmt.Sprintf("%s:/data:Z", tmpDir),
 		cfg.containerImage,
 		"--yaml-root", path.Join("/data/input/semantic_conventions/", path.Base(cfg.inputPath)),
 	}

